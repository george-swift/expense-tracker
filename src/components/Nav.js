import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Offcanvas from 'react-bootstrap/Offcanvas';
import Fab from '@mui/material/Fab';
import HomeIcon from '@mui/icons-material/Home';
import NavigationIcon from '@mui/icons-material/Navigation';
import Nav from 'react-bootstrap/Nav';
import DonutLargeTwoToneIcon from '@mui/icons-material/DonutLargeTwoTone';
import ManageAccountsTwoToneIcon from '@mui/icons-material/ManageAccountsTwoTone';

const Navbar = ({ show, handleShow, handleClose }) => (
  <div className="navigate px-3">
    <Fab variant="extended" sx={{ color: '#8f659a' }} onClick={handleShow}>
      <NavigationIcon sx={{ mr: 0.5 }} />
      Navigate
    </Fab>

    <Offcanvas show={show} onHide={handleClose} placement="end">
      <Offcanvas.Header closeButton>
        <Offcanvas.Title>Navigate</Offcanvas.Title>
      </Offcanvas.Header>
      <Offcanvas.Body>
        <Nav className="flex-column mt-4">
          <Nav.Item className="mb-3">
            <Link className="nav-link" onClick={handleClose} to="/app">
              <HomeIcon sx={{ mr: 2 }} />
              Home
            </Link>
          </Nav.Item>
          <Nav.Item className="mb-3">
            <Link className="nav-link" onClick={handleClose} to="/app/reports">
              <DonutLargeTwoToneIcon sx={{ mr: 2 }} />
              Reports
            </Link>
          </Nav.Item>
          <Nav.Item>
            <Link className="nav-link" onClick={handleClose} to="/app/more">
              <ManageAccountsTwoToneIcon sx={{ mr: 2 }} />
              Edit Profile
            </Link>
          </Nav.Item>
        </Nav>
      </Offcanvas.Body>
    </Offcanvas>
  </div>
);

Navbar.propTypes = {
  show: PropTypes.bool.isRequired,
  handleShow: PropTypes.func.isRequired,
  handleClose: PropTypes.func.isRequired,
};

export default Navbar;
