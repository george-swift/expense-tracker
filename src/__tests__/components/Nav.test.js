import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Nav from '../../components/Nav';

describe('Testing the Nav component', () => {
  it('should render correctly at the bottom of page', () => {
    const tree = renderer
      .create(
        <Router>
          <Nav />
        </Router>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
