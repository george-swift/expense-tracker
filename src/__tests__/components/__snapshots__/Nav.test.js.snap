// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Nav component should render correctly at the bottom of page 1`] = `
<div
  className="app-nav px-3"
>
  <a
    aria-controls="offcanvas"
    className="trigger"
    data-bs-toggle="offcanvas"
    href="#offcanvas"
    role="button"
  >
    Navigation Links
    <span
      className="ms-2"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M272 464a16 16 0 01-16-16.42V264.13a8 8 0 00-8-8H64.41a16.31 16.31 0 01-15.49-10.65 16 16 0 018.41-19.87l384-176.15a16 16 0 0121.22 21.19l-176 384A16 16 0 01272 464z"
        />
      </svg>
    </span>
  </a>
  <div
    aria-labelledby="offcanvasExample"
    className="offcanvas offcanvas-bottom bg-purple"
    id="offcanvas"
    tabIndex="-1"
  >
    <div
      className="offcanvas-header"
    >
      <h5
        className="offcanvas-title fw-bold"
      >
        Click links below to navigate
      </h5>
      <button
        aria-label="Close"
        className="btn-close text-reset"
        data-bs-dismiss="offcanvas"
        type="button"
      />
    </div>
    <div
      className="offcanvas-body"
    >
      <ul
        className="nav flex-column px-0"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/et"
            onClick={[Function]}
          >
            <span
              className="p-1 me-3"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M80 368H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm0-320H16A16 16 0 0 0 0 64v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16V64a16 16 0 0 0-16-16zm0 160H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm416 176H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z"
                />
              </svg>
            </span>
            Manage
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/et/reports"
            onClick={[Function]}
          >
            <span
              className="p-1 me-3"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 544 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M527.79 288H290.5l158.03 158.03c6.04 6.04 15.98 6.53 22.19.68 38.7-36.46 65.32-85.61 73.13-140.86 1.34-9.46-6.51-17.85-16.06-17.85zm-15.83-64.8C503.72 103.74 408.26 8.28 288.8.04 279.68-.59 272 7.1 272 16.24V240h223.77c9.14 0 16.82-7.68 16.19-16.8zM224 288V50.71c0-9.55-8.39-17.4-17.84-16.06C86.99 51.49-4.1 155.6.14 280.37 4.5 408.51 114.83 513.59 243.03 511.98c50.4-.63 96.97-16.87 135.26-44.03 7.9-5.6 8.42-17.23 1.57-24.08L224 288z"
                />
              </svg>
            </span>
            Reports
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/et/more"
            onClick={[Function]}
          >
            <span
              className="p-1 me-3"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M12 14v8H4a8 8 0 0 1 8-8zm0-1c-3.315 0-6-2.685-6-6s2.685-6 6-6 6 2.685 6 6-2.685 6-6 6zm2.595 5.812a3.51 3.51 0 0 1 0-1.623l-.992-.573 1-1.732.992.573A3.496 3.496 0 0 1 17 14.645V13.5h2v1.145c.532.158 1.012.44 1.405.812l.992-.573 1 1.732-.992.573a3.51 3.51 0 0 1 0 1.622l.992.573-1 1.732-.992-.573a3.496 3.496 0 0 1-1.405.812V22.5h-2v-1.145a3.496 3.496 0 0 1-1.405-.812l-.992.573-1-1.732.992-.572zM18 17a1 1 0 1 0 0 2 1 1 0 0 0 0-2z"
                  />
                </g>
              </svg>
            </span>
            Edit Profile
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
